crates/sui-rosetta/src/operations.rs

```diff
@@ -558,6 +558,74 @@ impl Operations {
         };
         balance_change.chain(gas)
     }
+
+    fn process_single_gascoin_transfer(
+        coin_change_operations: &mut impl Iterator<Item = crate::operations::Operation>,
+        tx: SuiTransactionBlockKind,
+        sender: SuiAddress,
+        gas_used: i128,
+    ) -> Vec<Operation> {
+        let mut is_single_gascoin_transfer = false;
+        match tx {
+            SuiTransactionBlockKind::ProgrammableTransaction(pt) => {
+                let SuiProgrammableTransactionBlock {
+                    inputs: _,
+                    commands,
+                } = &pt;
+                for command in commands {
```

Comment:
In case we find a command with gas-coin we can stop iterating.
Maybe something like `commands.into_iter().find(|c| { ... })` would be more suitable.

Reply:
Done